common/components/common/projects/ProjectVolunteerModal.jsx

```diff
@@ -96,7 +96,12 @@ class ProjectVolunteerModal extends React.PureComponent<Props, State> {
   }
 
   onRoleChange(role: $ReadOnlyArray<TagDefinition>): void {
-    this.state.roleTag = role[0];
+    //this fixes an Uncaught TypeError: Cannot read property '0' of null if the user clicks the X to clear the field in the role selector
```

Comment:
I don't think this was added by my branch but I don't like console errors and it was a quick fix. when role was set to undefined or null, this component attempted to read null[0] which didn't work. Now it checks for that before setting state.

Reply:
Cool, you could make it a one-liner with 
```suggestion
   this.state.roleTag = !_.isEmpty(role) ? role[0] : null;
```

Reply:
Just made this change. Good catch, thanks!